elytron/WFCORE-5178_native-support-for-oidc.adoc

```diff
@@ -0,0 +1,164 @@
+
+==== Token storage
+
+The new HTTP mechanism implementation will be able to make use of a session token store and a cookie token store
+to store account info (i.e., access tokens, ID tokens, and refresh tokens).
+
+The session token store will be used by default but it should be possible to specify that the cookie token store
+should be used instead.
+
+The new HTTP mechanism implementation will refresh the access token when it is expired. Since the Keycloak client adapter
+provides an option to refresh the token on every request, we should provide this option as well but make
+sure to document that the use of this option is discouraged since it can have a significant impact on performance.
+
+==== New elytron-oidc subsystem
+
+A new subsystem, `elytron-oidc` will be added to WildFly. Its schema will be similar to the existing
+https://github.com/keycloak/keycloak/blob/master/adapters/oidc/wildfly/wildfly-subsystem/src/main/resources/schema/wildfly-keycloak_1_1.xsd[keycloak-adapter-subsystem].
+The new subsystem will install a `DeploymentUnitProcessor` which will detect if OIDC is required for a deployment.
+
+Like with the existing Keycloak client adapter, the OIDC configuration will be specified via the deployment
+or via subsystem configuration. For the deployment case, the OIDC configuration will be specified in a JSON
+configuration file in the WEB-INF directory along with the `auth-method`, `OIDC`, specified in the `web.xml`.
+For the subsystem case, the `auth-method` won't need to be specified in the `web.xml` file, it will be set
+automatically.
+
+Once a deployment has been processed, if its `auth-method` is `OIDC`, the OIDC integration will be activated.
+Otherwise, no activation will occur and the deployment will continue to use the existing security capabilities
+provided by WildFly.
+
+If the OIDC integration is activated for a deployment, then appropriate module dependencies will be added to
+the deployment (e.g., a dependency on the new Elytron OIDC module). This will ensure the activation of the
+OIDC HTTP mechanism implementation for the deployment. The deployment will make use of a "Virtual Security Domain",
+similar to what was implemented for https://github.com/wildfly/wildfly-proposals/blob/master/microprofile/WFLY-12609_MicroProfile_JWT_1_1.adoc#hard-requirements[Microprofile JWT].
+This means that it will no longer be necessary to pre-configure SecurityDomains in the Elytron subsystem,
+as was required with the Keycloak client adapter.
+
+It should be possible to make use of different OpenID providers with WildFly. While working on the
+implementation for this RFE, we should make sure that we aren't including Keycloak-specific configuration, if
+possible.
+
+==== New galleon layer for OIDC support
+
+A new Galleon layer, `oidc`, should be added to allow a server to be provisioned with OIDC support.
+
+=== Nice-to-Have Requirements
```

Comment:
The authentication and authorization is only for deployment or can be used also for server management?

Reply:
This is for securing applications that are deployed to the server. Securing the admin console will require some additional work so that might be done as part of a follow-up task.